opte/src/engine/packet.rs

```diff
@@ -1433,9 +1559,13 @@ impl Packet<Parsed> {
                             .unwrap();
                         let pseudo_csum = ip.pseudo_csum();
                         csum += pseudo_csum;
-                        csum.add(
-                            self.segs[0].slice(tcp_off.seg_pos, tcp.hdr_len()),
-                        );
+                        // Safety: We are slicing with the parse data.
+                        csum.add(unsafe {
```

Comment:
Why use the unsafe / unchecked method here?

Reply:
Because it's the only option available here, we don't know at compile time that the offset is valid (though practically speaking we know it's fine because it came from the parsed data). Anyways, this is moot now. I changed these functions to no longer be unsafe, but instead panic if you slice out of bound of the segment.